### PR TITLE
Update resharding config section

### PR DESCRIPTION
**Pull request by nickva**
Some entries were stale and others were missing

### CODE REVIEW

File: src/config/resharding.rst [hunk 1]
```diff
@@ -32,7 +32,17 @@ Resharding Configuration
         will start to fail with the ``max_jobs_exceeded`` error::
 
              [reshard]
-             max_jobs = 25
+             max_jobs = 48
+
+    .. config:option:: max_history
+
+        The max length of each jobs event log. The history records job starts,
+        stop, retries etc. There is probably not much reason to change this
```
Comment (flimzy) [inline src/config/resharding.rst]:
```suggestion
        stops, retries etc. There is probably not much reason to change this
```

File: src/config/resharding.rst [hunk 2]
```diff
@@ -32,7 +32,17 @@ Resharding Configuration
...
+        unless use specifies larger number of ``max_retries``::
```
Comment (flimzy) [inline src/config/resharding.rst]: I'm not sure what is meant here.  Maybe this?

```suggestion
        unless you specify a larger number of ``max_retries``::
```

File: src/config/resharding.rst [hunk 3]
```diff
@@ -77,3 +87,22 @@ Resharding Configuration
...
+        jobs. This can be used as a safety check to avoid inadvertently
+        starting too many resharding jobs by accident::
+
+
+            [reshard]
+            require_node_param = false
+
+    .. config:option:: require_range_param
+
+        Require users to specify a ``node`` parameter when creating resharding
```
Comment (flimzy) [inline src/config/resharding.rst]: This looks like a duplicate of the last paragraph. Wrong description?

File: src/config/resharding.rst (hunk 2 above)
Comment (nickva) [reply src/config/resharding.rst]: I think I'll remove that sentence. It doesn't make much sense in general.

File: src/config/resharding.rst (hunk 3 above)
Comment (nickva) [reply src/config/resharding.rst]: It was a copy-paste error. It was supposed to be range instance of node. I'll update it.

**Pull request merged**